ts/Stock/Indicators/ATR/ATRIndicator.ts

```diff
@@ -0,0 +1,250 @@
+/* *
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import type IndicatorValuesObject from '../IndicatorValuesObject';
+import type LineSeries from '../../../Series/Line/LineSeries';
+import type {
+    ATROptions,
+    ATRParamsOptions
+} from './ATROptions';
+import type ATRPoint from './ATRPoint';
+import BaseSeries from '../../../Core/Series/Series.js';
+const {
+    seriesTypes: {
+        sma: SMAIndicator
+    }
+} = BaseSeries;
+import U from '../../../Core/Utilities.js';
+const {
+    isArray,
+    merge
+} = U;
+
+/* eslint-disable valid-jsdoc */
+// Utils:
+
+/**
+ * @private
+ */
+function accumulateAverage(
+    points: Array<[number, Array<number>]>,
+    xVal: Array<number>,
+    yVal: Array<Array<number>>,
+    i: number
+): void {
+    var xValue = xVal[i],
+        yValue = yVal[i];
+
+    points.push([xValue, yValue]);
+}
+
+/**
+ * @private
+ */
+function getTR(
+    currentPoint: Array<number>,
+    prevPoint: Array<number>
+): number {
+    var pointY = currentPoint,
+        prevY = prevPoint,
+        HL = pointY[1] - pointY[2],
+        HCp = typeof prevY === 'undefined' ? 0 : Math.abs(pointY[1] - prevY[3]),
+        LCp = typeof prevY === 'undefined' ? 0 : Math.abs(pointY[2] - prevY[3]),
+        TR = Math.max(HL, HCp, LCp);
+
+    return TR;
+}
+
+/**
+ * @private
+ */
+function populateAverage(
+    points: Array<[number, Array<number>]>,
+    xVal: Array<number>,
+    yVal: Array<Array<number>>,
+    i: number,
+    period: number,
+    prevATR: number
+): Array<number> {
+    var x = xVal[i - 1],
+        TR = getTR(yVal[i - 1], yVal[i - 2]),
+        y;
+
+    y = (((prevATR * (period - 1)) + TR) / period);
+
+    return [x, y];
+}
+
+/* eslint-enable valid-jsdoc */
+
+/* *
+ *
+ * Class
+ *
+ * */
+
+/**
+ * The ATR series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.atr
+ *
+ * @augments Highcharts.Series
+ */
+class ATRIndicator extends SMAIndicator {
+    /**
+     * Average true range indicator (ATR). This series requires `linkedTo`
+     * option to be set.
+     *
+     * @sample stock/indicators/atr
+     *         ATR indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        6.0.0
+     * @product      highstock
+     * @requires     stock/indicators/indicators
+     * @requires     stock/indicators/atr
+     * @optionparent plotOptions.atr
+     */
+    public static defaultOptions: ATROptions = merge(SMAIndicator.defaultOptions, {
+        params: {
+            period: 14
+        }
+    } as ATROptions);
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public data: Array<ATRPoint> = void 0 as any;
+    public points: Array<ATRPoint> = void 0 as any;
```

Comment:
`options` is missing here.

Reply:
In the interface, I was missed.

Fixed.